### DISCUSSION TITLE
Terrain layer opacity

### DISCUSSION THREAD

**Discussion by radoslavirha** [visgl/deck.gl#7806] [Q&A] [answered]
Hello

I found that the opacity in TerrainLayer has changed (I don't know which was the last working version).

Currently if I have some underground data and change the opacity, I can't see anything under the surface (I'm pretty sure it used to work, I haven't found any changes in my code that would affect it).

I know I can try a `depthTest` based on `opacity`, but it looks weird, it's not blacked out like with semi-transparency.

Did something change with TerrainLayer? 

Turn on TerrainLayer, change opacity, turn off:

https://user-images.githubusercontent.com/23722745/228861511-176624e2-d6dc-4451-88d1-fe17ffbcc38a.mov

Thanks

**Comment by radoslavirha** [comment]
I found some older video


https://user-images.githubusercontent.com/23722745/228899214-c725ed17-5996-4ca7-bdca-2b213f58912b.mov

**Comment by Pessimistress** [comment, marked as answer]
You should render whatever layer is behind the terrain first, then the translucent terrain layer. Otherwise it will be blocked by depth test.
